### PR TITLE
[fix]: [CDS-52198]: handle deleted envgroup for multi env deployment

### PR DESCRIPTION
**Pull request by tchaurasiya**
## Describe your changes
* adding validation for for environment list /access api, if requested envGroup is missing
* adding validation in step too for the same,  considering case of pipeline execution through api/trigger.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43619)

### CODE REVIEW

**Comment by yogesh-chauhan** [comment]
@tchaurasiya lets update the PR description

**Review by RajBaviskar — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43619#-:-NNiAG8kBUJX3qyI6Sol:bnfp4nl)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by RajBaviskar — DISMISSED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43619)* status: 3 file(s) to review. 1 of 2 approvals obtained.

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steps/MultiDeploymentSpawnerStep.java [hunk 1]
```diff
@@ -100,9 +104,19 @@
...
       }
       if (isNotBlank(envGroupRef.getValue())) {
-        IdentifierRef envGroupIdentifierRef =
-            IdentifierRefHelper.getIdentifierRef(envGroupRef.getValue(), AmbianceUtils.getAccountId(ambiance),
-                AmbianceUtils.getOrgIdentifier(ambiance), AmbianceUtils.getProjectIdentifier(ambiance));
+        String accountIdentifier = AmbianceUtils.getAccountId(ambiance);
+        String orgIdentifier = AmbianceUtils.getOrgIdentifier(ambiance);
+        String projectIdentifier = AmbianceUtils.getProjectIdentifier(ambiance);
+
+        Optional<EnvironmentGroupEntity> environmentGroupEntity = environmentGroupService.get(
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steps/MultiDeploymentSpawnerStep.java:111]: does this take care of account/org level env groups ?

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

**Review by tchaurasiya — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43619)* status: 3 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steps/MultiDeploymentSpawnerStep.java` line 111 at r2](https://reviewable.io/reviews/harness/harness-core/43619#-NNftaldSxepTebsNvcq-r2-111:-NNlcYGp0ivcXaWcdJ69:b-7xt2v2) ([raw file](https://github.com/harness/harness-core/blob/4272e86c14cd402a638bf0224fc5ea0ec3ccf88a/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/pipeline/steps/MultiDeploymentSpawnerStep.java#L111)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

does this take care of account/org level env groups ?
</blockquote></details>

Yes!!

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43619)* status: :shipit: complete! ✓

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43619)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Comment by tchaurasiya** [comment]
trigger messagecheck

**Comment by tchaurasiya** [comment]
trigger messagecheck

**Review by utkarshchoubey-harness — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43619)* status: :shipit: complete! ✓

**Pull request merged**